share/modules/nixos-shell-config.nix

```
@@ -112,8 +112,8 @@ in {
               path = builtins.getEnv "PATH";
             in
             ''
-              # fix terminal size
-              eval "$(resize)"
+              # if terminal with stdout, fix terminal size
+              ${pkgs.coreutils}/bin/test '!' -t 1 || eval "$(resize)"
```

This should also works with posix and works with `set -e` (no idea who uses this in login shells ;)):

```suggestion
              if [ -t 1 ]; then eval "$(resize)"; fi
```

---

done.

I've briefly checked posix; test is also part of POSIX (but not an sh builtin), as well as `[`.
I'm not sure what this means in the context of NixOS (can there be a nixos without coreutils?).

but idc, the version is more readable.